app/controllers/api/v01/bibs_controller.rb

```diff
@@ -8,23 +8,50 @@ class Api::V01::BibsController < ApplicationController
   def create_or_update_teacher_sets
     error_code_and_message = validate_request
     if error_code_and_message.any?
+      AdminMailer.failed_bibs_controller_api_request(@request_body, error_code_and_message, action_name)
       render_error(error_code_and_message)
-      return
     end
+    return if error_code_and_message.any?
 
     saved_teacher_sets = []
     @request_body.each do |teacher_set_record|
-      teacher_set = TeacherSet.where(bnumber: "b#{teacher_set_record['id']}").first_or_initialize
+      # overwrite @teacher_set_record instance variable so it can be read in the var_field and fixed_field methods
+      @teacher_set_record = teacher_set_record
```

Comment:
Why add a "@" in front of teacher_set_record? Are you trying to access it globally from somewhere else?

Reply:
@nnat425 yes, I mentioned it in the comment above that line (might be hard to see):
"# overwrite @teacher_set_record instance variable so it can be read in the var_field and fixed_field methods"